src/resolve/resolved-varlink.c

```diff
@@ -534,6 +564,514 @@ static int vl_method_resolve_address(Varlink *link, JsonVariant *parameters, Var
         return 1;
 }
 
+static int append_txt(JsonVariant **_txt, DnsResourceRecord *rr) {
+        _cleanup_(json_variant_unrefp) JsonVariant *entry = NULL;
+        int r;
+
+        assert(rr);
+        assert(rr->key);
+
+        if (rr->key->type != DNS_TYPE_TXT)
+                return 0;
+
+        LIST_FOREACH(items, i, rr->txt.items) {
+                if (i->length <= 0)
+                        continue;
+
+                r = json_build(&entry,  JSON_BUILD_BYTE_ARRAY(i->data, i->length));
+                if (r < 0)
+                        return r;
+
+                r = json_variant_append_array(_txt, entry);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
this leaks memory, as you do not free the entry object here. i.e. move the declaration of `entry` into the loop